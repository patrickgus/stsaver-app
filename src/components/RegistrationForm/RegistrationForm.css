.RegistrationForm {
  display: flex;
  flex-direction: column;
  padding: 15px;
}

.RegistrationForm__buttons {
  display: flex;
  justify-content: space-evenly;
}
