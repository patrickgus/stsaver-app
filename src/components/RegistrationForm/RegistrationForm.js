import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import AuthApiService from "../../services/auth-api-service";
import "./RegistrationForm.css";

class RegistrationForm extends Component {
  state = { error: null };

  handleSubmit = e => {
    e.preventDefault();
    const { first_name, last_name, username, password } = e.target;

    this.setState({ error: null });
    AuthApiService.postUser({
      username: username.value,
      password: password.value,
      first_name: first_name.value,
      last_name: last_name.value
    })
      .then(user => {
        first_name.value = "";
        last_name.value = "";
        username.value = "";
        password.value = "";
        this.props.history.push("/activity");
      })
      .catch(res => {
        this.setState({ error: res.error });
      });
  };

  handleCancel = e => {
    e.preventDefault();
    this.props.history.push("/");
  };

  render() {
    const { error } = this.state;
    return (
      <form className="RegistrationForm" onSubmit={this.handleSubmit}>
        <div role="alert">{error && <p className="error">{error}</p>}</div>
        <div className="RegistrationForm__form-section">
          <label htmlFor="first_name">First name</label>
          <input name="first_name" type="text" required id="first_name" />
        </div>
        <div className="RegistrationForm__form-section">
          <label htmlFor="last_name">Last name</label>
          <input name="last_name" type="text" required id="last_name" />
        </div>
        <div className="RegistrationForm__form-section">
          <label htmlFor="username">Username</label>
          <input name="username" type="text" required id="username" />
        </div>
        <div className="RegistrationForm__form-section">
          <label htmlFor="password">Password</label>
          <input name="password" type="password" required id="password" />
        </div>
        <div className="RegistrationForm__buttons">
          <button type="submit">Register</button>
          <button onClick={this.handleCancel}>Cancel</button>
        </div>
      </form>
    );
  }
}

export default withRouter(RegistrationForm);
