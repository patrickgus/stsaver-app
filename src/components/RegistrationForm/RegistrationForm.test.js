import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import RegistrationForm from "./RegistrationForm";

describe("RegistrationForm component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <RegistrationForm />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the form as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <RegistrationForm />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
