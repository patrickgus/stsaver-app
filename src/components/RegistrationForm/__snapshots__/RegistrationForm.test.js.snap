// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm component renders the form as expected 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    className="RegistrationForm__form-section"
  >
    <label
      htmlFor="first_name"
    >
      First name
    </label>
    <input
      id="first_name"
      name="first_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="RegistrationForm__form-section"
  >
    <label
      htmlFor="last_name"
    >
      Last name
    </label>
    <input
      id="last_name"
      name="last_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="RegistrationForm__form-section"
  >
    <label
      htmlFor="username"
    >
      Username
    </label>
    <input
      id="username"
      name="username"
      required={true}
      type="text"
    />
  </div>
  <div
    className="RegistrationForm__form-section"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      id="password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <div
    className="RegistrationForm__buttons"
  >
    <button
      type="submit"
    >
      Register
    </button>
    <button
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</form>
`;
